Build failure: ffmpegthumbnailer on staging-next

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

nix shell nixpkgs/staging-next#ffmpegthumbnailer

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/312424868

### Relevant log output

```console
error: Cannot build '/nix/store/84idycdbj4a1dijrvh70vjwkih6vn9rr-ffmpegthumbnailer-unstable-2024-01-04.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/7yvriddaycwqnnxpb3gwj9y57pbpm550-ffmpegthumbnailer-unstable-2024-01-04
       Last 25 log lines:
       > [  5%] Building CXX object CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/moviedecoder.cpp.o
       > [ 11%] Building CXX object CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/stringoperations.cpp.o
       > [ 22%] Building CXX object CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/videothumbnailerc.cpp.o
       > [ 22%] Building CXX object CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/rgbwriter.cpp.o
       > [ 33%] Building CXX object CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/filmstripfilter.cpp.o
       > [ 33%] Building CXX object CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/videothumbnailer.cpp.o
       > [ 38%] Building CXX object CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/jpegwriter.cpp.o
       > [ 44%] Building CXX object CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/pngwriter.cpp.o
       > /build/source/libffmpegthumbnailer/moviedecoder.cpp: In member function 'void ffmpegthumbnailer::MovieDecoder::initializeFilterGraph(const AVRational&, const std::string&, bool)':
       > /build/source/libffmpegthumbnailer/moviedecoder.cpp:409:19: error: 'struct AVFrame' has no member named 'interlaced_frame'
       >   409 |     if (m_pFrame->interlaced_frame != 0)
       >       |                   ^~~~~~~~~~~~~~~~
       > /build/source/libffmpegthumbnailer/moviedecoder.cpp: In member function 'void ffmpegthumbnailer::MovieDecoder::seek(int)':
       > /build/source/libffmpegthumbnailer/moviedecoder.cpp:521:39: error: 'struct AVFrame' has no member named 'key_frame'
       >   521 |     } while ((!gotFrame || !m_pFrame->key_frame) && keyFrameAttempts < 200);
       >       |                                       ^~~~~~~~~
       > /build/source/libffmpegthumbnailer/moviedecoder.cpp: In member function 'int32_t ffmpegthumbnailer::MovieDecoder::getStreamRotation()':
       > /build/source/libffmpegthumbnailer/moviedecoder.cpp:658:46: error: 'av_stream_get_side_data' was not declared in this scope; did you mean 'av_frame_get_side_data'?
       >   658 |     auto matrix = reinterpret_cast<int32_t*>(av_stream_get_side_data(m_pVideoStream, AV_PKT_DATA_DISPLAYMATRIX, nullptr));
       >       |                                              ^~~~~~~~~~~~~~~~~~~~~~~
       >       |                                              av_frame_get_side_data
       > make[2]: *** [CMakeFiles/libffmpegthumbnailerobj.dir/build.make:79: CMakeFiles/libffmpegthumbnailerobj.dir/libffmpegthumbnailer/moviedecoder.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:119: CMakeFiles/libffmpegthumbnailerobj.dir/all] Error 2
       > make: *** [Makefile:166: all] Error 2
       For full logs, run:
         nix log /nix/store/84idycdbj4a1dijrvh70vjwkih6vn9rr-ffmpegthumbnailer-unstable-2024-01-04.drv
```

### Additional context



### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.7-cachyos, NixOS, 25.11 (Xantusia), 25.11.20251105.ffcdcf9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/ndig4f4dx8bmrmyr5vfm19g02r9l9ggm-source`

### Notify maintainers


@jagajaga 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ff/ffmpegthumbnailer/package.nix
+++ b/pkgs/by-name/ff/ffmpegthumbnailer/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  fetchpatch,
   cmake,
   pkg-config,
   ffmpeg-headless,
@@ -9,16 +10,24 @@
   libjpeg,
 }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation (finalAttrs: {
   pname = "ffmpegthumbnailer";
-  version = "unstable-2024-01-04";
+  version = "2.2.3";
 
   src = fetchFromGitHub {
     owner = "dirkvdb";
     repo = "ffmpegthumbnailer";
-    rev = "1b5a77983240bcf00a4ef7702c07bcd8f4e5f97c";
-    hash = "sha256-7SPRQMPgdvP7J3HCf7F1eXxZjUH5vCYZ9UOwTUFMLp0=";
+    tag = finalAttrs.version;
+    hash = "sha256-1hVPtCPwfovCtA6aagViUJkYTCFuiFkOqGEqMHIoZe8=";
   };
+
+  patches = [
+    (fetchpatch {
+      name = "ffmpeg-8-fix.patch";
+      url = "https://github.com/dirkvdb/ffmpegthumbnailer/commit/df789ec326ae0f2c619f91c8f2fc8b5e45b50a70.patch";
+      hash = "sha256-PArrcKuaWWA6/H59MbdC36B57GSvvp5sHz24QLTBZYw=";
+    })
+  ];
 
   nativeBuildInputs = [
     cmake
@@ -60,4 +69,4 @@ stdenv.mkDerivation {
     platforms = platforms.unix;
     mainProgram = "ffmpegthumbnailer";
   };
-}
+})

